---
title: "Radiomic features parallel RNA-seq analysis and enrichment"
output: pdf_document
author: "Carla Casanova"
date: '2022-06-13'
header-includes:
  - |
    ```{=latex}
    \usepackage{fvextra}
    \DefineVerbatimEnvironment{Highlighting}{Verbatim}{
      breaksymbolleft={}, 
      showspaces = false,
      showtabs = false,
      breaklines,
      commandchars=\\\{\}
    }
    ```
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE, cache = TRUE)
knitr::opts_chunk$set(tidy.opts = list(width.cutoff = 80), tidy = TRUE)
```


```{r libraries}
library(limma)
library(sva)
library(ggplot2)
library(org.Hs.eg.db)
library(annotate)
library(hgu133plus2.db)
library(clusterProfiler)
library(enrichplot)
library(httr)
library(cowplot)
```


This analysis will be performed with counts from **sputum samples**:  

```{r load_data}
load("/Users/carlacasanovasuarez/Documents/Master Bioinformatics UAB/Prácticas Radiomics/Radiomic features/Results_rfeatures/R objects/rdr_assay_scaled_combat.rda")
load("/Users/carlacasanovasuarez/Documents/Master Bioinformatics UAB/Prácticas Radiomics/Radiomic features/Results_rfeatures/R objects/sputum_eset_countsOK_Combat.rda")
load("/Users/carlacasanovasuarez/Documents/Master Bioinformatics UAB/Prácticas Radiomics/Radiomic features/Results_rfeatures/R objects/sputum_eset_phenoOK.rda")
load("/Users/carlacasanovasuarez/Documents/Master Bioinformatics UAB/Prácticas Radiomics/Radiomic features/Results_rfeatures/R objects/sputum_eset_annotationOK.rda")
```


```{r variables_indv}
variables_interest <- data.frame(Sex = as.factor(phenoDataFrame$SEX), 
                                 Age = phenoDataFrame$AGE, 
                                 Dwalk = phenoDataFrame$DWALK,
                                 BMI = phenoDataFrame$BMI,
                                 Smoke_history = phenoDataFrame$SUSMHS,
                                 Years_smoke = phenoDataFrame$SUSMYR,
                                 Cough = as.factor(phenoDataFrame$COUGH),
                                 ex_first_year = phenoDataFrame$Y1EXBS,
                                 Group = as.factor(phenoDataFrame$GROUP_bin),
                                 N_cigarrete_day = phenoDataFrame$SUCGSMDY,
                                 Cr_bronchitis = as.factor(phenoDataFrame$CBRONCH),
                                 Cr_wheezeng = as.factor(phenoDataFrame$ATS3EG),
                                 history_asthma = as.factor(phenoDataFrame$ATS5G),
                                 fume_expose = as.factor(phenoDataFrame$ATS6C),
                                 dusty_expose = as.factor(phenoDataFrame$ATS6B),
                                 phlegm = as.factor(phenoDataFrame$PHLEGM),
                                 heart_failure = as.factor(phenoDataFrame$ATS8F),
                                 stroke = as.factor(phenoDataFrame$ATS8E),
                                 diabetes = as.factor(phenoDataFrame$ATS8L),
                                 osteoporosis = as.factor(phenoDataFrame$ATS8H),
                                 FEVVCPD = phenoDataFrame$FEVVCPD,
                                 FEV1PSPC = phenoDataFrame$FEV1PSPC,
                                 TLC = phenoDataFrame$TLC,
                                 FRC = phenoDataFrame$FRC,
                                 Low_percentile = phenoDataFrame$LOW15PCT,
                                 row.names = rownames(phenoDataFrame))

# Code for removing columns with some NA
variables_interest <- variables_interest[ , apply(variables_interest, 2, function(x) !any(is.na(x)))]

# Add radiomic features without grouping them
identical(colnames(rdr_assay), rownames(variables_interest))
variables_interest <- cbind(variables_interest, t(rdr_assay))

## Select counts
countData <- counts.ok[rowSums(counts.ok) > 10, ]

# Check for variables with only 1 factor to remove them from the analysis
variables.1.factor <- sapply(lapply(variables_interest, unique), length)
names(variables.1.factor[variables.1.factor == 1])

# Remove variables with just one factor
variables_interest[,"Smoke_history"] <- NULL
```

```{r list_loop}
r.features <- rownames(rdr_assay)
# Features que faltan por calcular
#small.r.features <- r.features[25]

# Prepare data in object: variables_interest
names(variables_interest)[names(variables_interest) == "10Percentile.original"] <- "X10Percentile.original"
names(variables_interest)[names(variables_interest) == "90Percentile.original"] <- "X90Percentile.original"

# Prepare data in object: small.r.features
r.features[15] <- "X10Percentile.original"
r.features[16] <- "X90Percentile.original"
```

Compute analysis for individual features by iterating (code parallelize):  

```{r rf_indv_parallel, eval=FALSE}
library(parallel)

# Use the detectCores() function to find the number of cores in system
no_cores <- detectCores()

# Setup cluster: good practice not to use all the cores
# Don't use all the cores at ISGlobal's cluster!!
clust <- makeCluster(no_cores-3) #This line will take time

# Export objects from global environment
clusterExport(cl = clust, varlist = c("variables_interest", "countData", "r.features"))

# Export packages to use
clusterEvalQ(clust, {
  library(tidyverse)
  library(limma)
  library(sva)
})


de_radiomic_features_indv <- function(x) {
  
  # Null formula
  # Group + Dwalk + FEV1PSPC + fume_expose + dusty_expose + history_asthma + Cough + BMI + Cr_wheezeng
  m.model <- "~ Sex + Age + BMI + Years_smoke"
  
  # -------SVA------ 
  # Create a model with the feature and a null model
  mod1 <- model.matrix(formula(paste(m.model, x, sep = "+")), data=variables_interest) 
  mod0 <- model.matrix(formula(m.model), data=variables_interest)
  
  # This estimates the required number of surrogate variables
  res <- svaseq(countData, mod1, mod0)
  # Add SVs to column data information
  variables_interest_sva <- data.frame(variables_interest, res$sv)
  
  # Add sva names to formula
  sva.names <- colnames(variables_interest_sva[,(ncol(variables_interest_sva)+1-res$n.sv):ncol(variables_interest_sva)])
  sva.collapsed <- paste(sva.names, collapse = "+")
  
  # -------VOOM-------
  # Normalize data with voom and add tumor stage as covariate in design 
  design <- model.matrix(formula(paste(m.model, x, sva.collapsed, sep = "+")), data=variables_interest_sva)
  v <- voom(countData, design=design, plot=FALSE)
  
  # fit the model
  fit <- lmFit(v, design)
  fit <- eBayes(fit)
  
  # Store the object
  save(fit, file = paste("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/", x, ".rda", sep = ""))
}

# With parallel
system.time(result <- parLapply(clust,r.features, de_radiomic_features_indv))
# Good practise to do that at the end
stopCluster(clust)
```

## Results DE of individual radiomic features  

Check summaries of voom results for each feature:  

```{r load_results_elongation}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/Elongation.original.rda")

summa.fit.elongation <- decideTests(fit)

summary(summa.fit.elongation)
```

```{r load_results_flatness}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/Flatness.original.rda")

summa.fit.flatness <- decideTests(fit)
summary(summa.fit.flatness)
```


```{r load_results_leastAxis}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/LeastAxisLength.original.rda")

summa.fit.leastAxis <- decideTests(fit)
summary(summa.fit.leastAxis)
```


```{r load_results_majorAxis}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/MajorAxisLength.original.rda")

summa.fit.majorAxis <- decideTests(fit)
summary(summa.fit.majorAxis)
```


```{r load_results_2ddCol}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/Maximum2DDiameterColumn.original.rda")

summa.fit.2ddCol <- decideTests(fit)
summary(summa.fit.2ddCol)
```


```{r load_results_2ddrow}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/Maximum2DDiameterRow.original.rda")

summa.fit.2ddRow <- decideTests(fit)
summary(summa.fit.2ddRow)
```


```{r load_results_2ddSlic}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/Maximum2DDiameterSlice.original.rda")

summa.fit.2ddSlic <- decideTests(fit)
summary(summa.fit.2ddSlic)
```


```{r load_results_3ddDia}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/Maximum3DDiameter.original.rda")

summa.fit.3ddDia <- decideTests(fit)
summary(summa.fit.3ddDia)
```


```{r load_results_MeshVol}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/MeshVolume.original.rda")

# Store: toptable, p.value, summary
summa.fit.MeshVol <- decideTests(fit)
toptable.MeshVol <- topTable(fit, coef="MeshVolume.original", number = dim(counts.ok)[1])
toptable.MeshVol <- toptable.MeshVol[order(toptable.MeshVol$P.Value),]
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = MeshVolume.original)) +
geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$MeshVolume.original)

summary(summa.fit.MeshVol)
head(toptable.MeshVol, 10)
```


```{r load_results_MinAxis}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/MinorAxisLength.original.rda")

summa.fit.MinAxis <- decideTests(fit)
summary(summa.fit.MinAxis)
```


```{r load_results_sphe}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/Sphericity.original.rda")

# Store: toptable, p.value, summary
summa.fit.sphe <- decideTests(fit)
toptable.sphericity <- topTable(fit, coef="Sphericity.original", number = dim(counts.ok)[1])
toptable.sphericity <- toptable.sphericity[order(toptable.sphericity$P.Value),]
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = Sphericity.original)) +
geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$Sphericity.original)

summary(summa.fit.sphe)
head(toptable.sphericity, 10)
```


```{r load_results_surAr}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/SurfaceArea.original.rda")

summa.fit.surAr <- decideTests(fit)
summary(summa.fit.surAr)
```


```{r load_results_surVol}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/SurfaceVolumeRatio.original.rda")

summa.fit.surVol <- decideTests(fit)
summary(summa.fit.surVol)
```


```{r load_results_voxVol}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/VoxelVolume.original.rda")

# Store: toptable, p.value, summary
summa.fit.voxVol <- decideTests(fit)
toptable.voxVol <- topTable(fit, coef="VoxelVolume.original", number = dim(counts.ok)[1])
toptable.voxVol <- toptable.voxVol[order(toptable.voxVol$P.Value),]
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = VoxelVolume.original)) +
geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$VoxelVolume.original)

summary(summa.fit.voxVol)
head(toptable.voxVol, 10)
```


```{r load_results_totEn}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/TotalEnergy.original.rda")

summa.fit.totEn <- decideTests(fit)
summary(summa.fit.totEn)
```


```{r load_results_unif}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/Uniformity.original.rda")

summa.fit.unif <- decideTests(fit)
summary(summa.fit.unif)
```


```{r load_results_var}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/Variance.original.rda")

# Store toptable, p-value, summary
summa.fit.var <- decideTests(fit)
toptable.variance <- topTable(fit, coef="Variance.original", number = dim(counts.ok)[1])
toptable.variance <- toptable.variance[order(toptable.variance$P.Value),]
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = Variance.original)) +
geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$Variance.original)

summary(summa.fit.var)
head(toptable.variance, 10)
```


```{r load_results_auto}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/Autocorrelation.original.rda")

# Store toptable, p.value, summary
summa.fit.auto <- decideTests(fit)
toptable.autoc <- topTable(fit, coef="Autocorrelation.original", number = dim(counts.ok)[1])
toptable.autoc <- toptable.autoc[order(toptable.autoc$P.Value),]
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = Autocorrelation.original)) +
geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$Autocorrelation.original)

summary(summa.fit.auto)
head(toptable.autoc, 10)
```


```{r load_results_clustPro}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/ClusterProminence.original.rda")

summa.fit.clustPro <- decideTests(fit)
summary(summa.fit.clustPro)
```


```{r load_results_clustSh}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/ClusterShade.original.rda")

summa.fit.clustSh <- decideTests(fit)
toptable.clustSh <- topTable(fit, coef="ClusterShade.original", number = dim(counts.ok)[1])
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = ClusterShade.original)) +
geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$ClusterShade.original)

summary(summa.fit.clustSh)
head(toptable.clustSh, 10)
```


```{r load_results_clustTen}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/ClusterTendency.original.rda")

summa.fit.clustTen <- decideTests(fit)
summary(summa.fit.clustTen)
```


```{r load_results_cont}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/Contrast.original.rda")

summa.fit.cont <- decideTests(fit)
summary(summa.fit.cont)
```


```{r load_results_corr}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/Maximum2DDiameterColumn.original.rda")

summa.fit.corr <- decideTests(fit)
summary(summa.fit.corr)
```


```{r load_results_diffAv}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/DifferenceAverage.original.rda")

summa.fit.diffAv <- decideTests(fit)
summary(summa.fit.diffAv)
```


```{r load_results_diffEnt}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/DifferenceEntropy.original.rda")

summa.fit.diffEnt <- decideTests(fit)
summary(summa.fit.diffEnt)
```


```{r load_results_diffVar}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/DifferenceVariance.original.rda")

summa.fit.diffVar <- decideTests(fit)
summary(summa.fit.diffVar)
```


```{r load_results_idO}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/Id.original.rda")

summa.fit.idO <- decideTests(fit)
summary(summa.fit.idO)
```


```{r load_results_idmO}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/Idm.original.rda")

summa.fit.idmO <- decideTests(fit)
summary(summa.fit.idmO)
```


```{r load_results_idmnO}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/Idmn.original.rda")

# Store toptable, p.value, summary
summa.fit.idmnO <- decideTests(fit)
summary(summa.fit.idmnO)
```


```{r load_results_idnO}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/Idn.original.rda")

summa.fit.idnO <- decideTests(fit)
summary(summa.fit.idnO)
```


```{r load_results_invV}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/InverseVariance.original.rda")

summa.fit.invV <- decideTests(fit)
summary(summa.fit.invV)
```


```{r load_results_jAv}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/JointAverage.original.rda")

summa.fit.jAv <- decideTests(fit)
toptable.jAv <- topTable(fit, coef="JointAverage.original", number = dim(counts.ok)[1])
toptable.jAv <- toptable.jAv[order(toptable.jAv$P.Value),]
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = JointAverage.original)) + 
  geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$JointAverage.original)

summary(summa.fit.jAv)
head(toptable.jAv, 10)
```


```{r load_results_jEn}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/JointEnergy.original.rda")

summa.fit.jEn <- decideTests(fit)
summary(summa.fit.jEn)
```


```{r load_results_jEnt}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/JointEntropy.original.rda")

summa.fit.jEnt <- decideTests(fit)
summary(summa.fit.jEnt)
```


```{r load_results_mcc}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/MCC.original.rda")

summa.fit.mcc <- decideTests(fit)
summary(summa.fit.mcc)
```


```{r load_results_maxProb}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/MaximumProbability.original.rda")

summa.fit.maxProb <- decideTests(fit)
summary(summa.fit.maxProb)
```


```{r load_results_sumAv}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/SumAverage.original.rda")

summa.fit.sumAv <- decideTests(fit)
toptable.sumAv <- topTable(fit, coef="SumAverage.original", number = dim(counts.ok)[1])
toptable.sumAv <- toptable.sumAv[order(toptable.sumAv$P.Value),]
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = SumAverage.original)) + 
  geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$SumAverage.original)

summary(summa.fit.sumAv)
head(toptable.sumAv, 10)
```


```{r load_results_sumEnt}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/SumEntropy.original.rda")

summa.fit.sumEnt <- decideTests(fit)
summary(summa.fit.sumEnt)
```


```{r load_results_sumSq}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/SumSquares.original.rda")

summa.fit.sumSq <- decideTests(fit)
summary(summa.fit.sumSq)
```


```{r load_results_glNU}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/GrayLevelNonUniformity.original.rda")

summa.fit.glNU <- decideTests(fit)
summary(summa.fit.glNU)
```


```{r load_results_glNUn}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/GrayLevelNonUniformityNormalized.original.rda")

summa.fit.glNUn <- decideTests(fit)
summary(summa.fit.glNUn)
```


```{r load_results_glVar}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/GrayLevelVariance.original.rda")

summa.fit.glVar <- decideTests(fit)
summary(summa.fit.glVar)
```


```{r load_results_hglEm}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/HighGrayLevelEmphasis.original.rda")

summa.fit.hglEm <- decideTests(fit)
toptable.hglEm <- topTable(fit, coef="HighGrayLevelEmphasis.original", number = dim(counts.ok)[1])
toptable.hglEm <- toptable.hglEm[order(toptable.hglEm$P.Value),]
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = HighGrayLevelEmphasis.original)) + 
  geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$HighGrayLevelEmphasis.original)

summary(summa.fit.hglEm)
head(toptable.hglEm, 10)
```


```{r load_results_lrEm}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/LongRunEmphasis.original.rda")

summa.fit.lrEm <- decideTests(fit)
summary(summa.fit.lrEm)
```


```{r load_results_lrHem}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/LongRunHighGrayLevelEmphasis.original.rda")

summa.fit.lrHem <- decideTests(fit)
summary(summa.fit.lrHem)
```


```{r load_results_lrLem}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/LongRunLowGrayLevelEmphasis.original.rda")

summa.fit.lrLem <- decideTests(fit)
toptable.lrLEm <- topTable(fit, coef="LongRunLowGrayLevelEmphasis.original", number = dim(counts.ok)[1])
toptable.lrLEm <- toptable.lrLEm[order(toptable.lrLEm$P.Value),]
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = LongRunLowGrayLevelEmphasis.original)) + 
  geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$LongRunLowGrayLevelEmphasis.original)

summary(summa.fit.lrLem)
head(toptable.lrLEm, 10)
```


```{r load_results_lowGLRem}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/LowGrayLevelRunEmphasis.original.rda")

summa.fit.lowGLRem <- decideTests(fit)
summary(summa.fit.lowGLRem)
```


```{r load_results_runEnt}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/RunEntropy.original.rda")

summa.fit.runEnt <- decideTests(fit)
summary(summa.fit.runEnt)
```


```{r load_results_rlNU}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/RunLengthNonUniformity.original.rda")

summa.fit.rlNU <- decideTests(fit)
summary(summa.fit.rlNU)
```


```{r load_results_rlNUn}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/RunLengthNonUniformityNormalized.original.rda")

summa.fit.rlNUn <- decideTests(fit)
summary(summa.fit.rlNUn)
```


```{r load_results_rPer}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/RunPercentage.original.rda")

summa.fit.rPer <- decideTests(fit)
summary(summa.fit.rPer)
```


```{r load_results_rVar}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/RunVariance.original.rda")

summa.fit.rVar <- decideTests(fit)
summary(summa.fit.rVar)
```


```{r load_results_shREm}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/ShortRunEmphasis.original.rda")

summa.fit.shREM <- decideTests(fit)
summary(summa.fit.shREM)
```


```{r load_results_shRH}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/ShortRunHighGrayLevelEmphasis.original.rda")

summa.fit.shRH <- decideTests(fit)
summary(summa.fit.shRH)
```


```{r load_results_shRL}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/ShortRunLowGrayLevelEmphasis.original.rda")

summa.fit.shRL <- decideTests(fit)
summary(summa.fit.shRL)
```


```{r load_results_hglZEM}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/HighGrayLevelZoneEmphasis.original.rda")

# Store toptable, p.value, summary
summa.fit.hglZEM <- decideTests(fit)
toptable.hglZEM <- topTable(fit, coef="HighGrayLevelZoneEmphasis.original", number = dim(counts.ok)[1])
toptable.hglZEM <- toptable.hglZEM[order(toptable.hglZEM$P.Value),]
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = HighGrayLevelZoneEmphasis.original)) +
geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$HighGrayLevelZoneEmphasis.original)

summary(summa.fit.hglZEM)
head(toptable.hglZEM, 10)
```


```{r load_results_larAEm}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/LargeAreaEmphasis.original.rda")

# Store toptable, p.value, summary
summa.fit.larAEm <- decideTests(fit)
toptable.larAEm <- topTable(fit, coef="LargeAreaEmphasis.original", number = dim(counts.ok)[1])
toptable.larAEm <- toptable.larAEm[order(toptable.larAEm$P.Value),]
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = LargeAreaEmphasis.original)) + 
  geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$LargeAreaEmphasis.original)

summary(summa.fit.larAEm)
head(toptable.larAEm, 10)
```


```{r load_results_lowGLem}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/LowGrayLevelEmphasis.original.rda")

# Store toptable, p.value, summary
summa.fit.lowGLem <- decideTests(fit)
toptable.lowGLem <- topTable(fit, coef="LowGrayLevelEmphasis.original", number = dim(counts.ok)[1])
toptable.lowGLem <- toptable.lowGLem[order(toptable.lowGLem$P.Value),]
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = LowGrayLevelEmphasis.original)) + 
  geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$LowGrayLevelEmphasis.original)

summary(summa.fit.lowGLem)
head(toptable.lowGLem, 10)
```


```{r load_results_larAHigh}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/LargeAreaHighGrayLevelEmphasis.original.rda")

# Store toptable, p.value, summary
summa.fit.larAHigh <- decideTests(fit)
toptable.larAHigh <- topTable(fit, coef="LargeAreaHighGrayLevelEmphasis.original", number = dim(counts.ok)[1])
toptable.larAHigh <- toptable.larAHigh[order(toptable.larAHigh$P.Value),]
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = LargeAreaHighGrayLevelEmphasis.original)) + 
  geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$LargeAreaHighGrayLevelEmphasis.original)

summary(summa.fit.larAHigh)
head(toptable.larAHigh, 10)
```


```{r load_results_larALow}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/LargeAreaLowGrayLevelEmphasis.original.rda")

# Store toptable, p.value, summary
summa.fit.larALow <- decideTests(fit)
toptable.larALow <- topTable(fit, coef="LargeAreaLowGrayLevelEmphasis.original", number = dim(counts.ok)[1])
toptable.larALow <- toptable.larALow[order(toptable.larALow$P.Value),]
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = LargeAreaLowGrayLevelEmphasis.original)) + 
  geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$LargeAreaLowGrayLevelEmphasis.original)

summary(summa.fit.larALow)
head(toptable.larALow)
```


```{r load_results_lowGLZem}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/LowGrayLevelZoneEmphasis.original.rda")

# Store toptable, p.value, summary
summa.fit.lowGLZem <- decideTests(fit)
toptable.lowGLZem <- topTable(fit, coef="LowGrayLevelZoneEmphasis.original", number = dim(counts.ok)[1])
toptable.lowGLZem <- toptable.lowGLZem[order(toptable.lowGLZem$P.Value),]
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = LowGrayLevelZoneEmphasis.original)) + 
  geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$LowGrayLevelZoneEmphasis.original)

summary(summa.fit.lowGLZem)
head(toptable.lowGLZem)
```


```{r load_results_sizZNU}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/SizeZoneNonUniformity.original.rda")

summa.fit.sizZNU <- decideTests(fit)
summary(summa.fit.sizZNU)
```


```{r load_results_sizZNUn}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/SizeZoneNonUniformityNormalized.original.rda")

summa.fit.sizZNUn <- decideTests(fit)
summary(summa.fit.sizZNUn)
```


```{r load_results_smAem}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/SmallAreaEmphasis.original.rda")

summa.fit.smAem <- decideTests(fit)
toptable.smAem <- topTable(fit, coef="SmallAreaEmphasis.original", number = dim(counts.ok)[1])
toptable.smAem <- toptable.smAem[order(toptable.smAem$P.Value),]
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = SmallAreaEmphasis.original)) + 
  geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$SmallAreaEmphasis.original)

summary(summa.fit.smAem)
head(toptable.smAem, 10)
```


```{r load_results_smAHem}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/SmallAreaHighGrayLevelEmphasis.original.rda")

summa.fit.smAHem <- decideTests(fit)
toptable.smAHem <- topTable(fit, coef="SmallAreaHighGrayLevelEmphasis.original", number = dim(counts.ok)[1])
toptable.smAHem <- toptable.smAHem[order(toptable.smAHem$P.Value),]
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = SmallAreaHighGrayLevelEmphasis.original)) + 
  geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$SmallAreaHighGrayLevelEmphasis.original)

summary(summa.fit.smAHem)
head(toptable.smAHem)
```


```{r load_results_smALem}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/SmallAreaLowGrayLevelEmphasis.original.rda")

summa.fit.smALem <- decideTests(fit)
toptable.smALem <- topTable(fit, coef="SmallAreaLowGrayLevelEmphasis.original", number = dim(counts.ok)[1])
toptable.smALem <- toptable.smALem[order(toptable.smALem$P.Value),]
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = SmallAreaLowGrayLevelEmphasis.original)) + 
  geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$SmallAreaLowGrayLevelEmphasis.original)

summary(summa.fit.smALem)
head(toptable.smALem, 10)
```


```{r load_results_zoneEn}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/ZoneEntropy.original.rda")

summa.fit.zoneEn <- decideTests(fit)
summary(summa.fit.zoneEn)
```


```{r load_results_zonePer}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/ZonePercentage.original.rda")

summa.fit.zonePer <- decideTests(fit)
summary(summa.fit.zonePer)
```


```{r load_results_zoneVar}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/ZoneVariance.original.rda")

summa.fit.zoneVar <- decideTests(fit)
toptable.zoneVar <- topTable(fit, coef="ZoneVariance.original", number = dim(counts.ok)[1])
toptable.zoneVar <- toptable.zoneVar[order(toptable.zoneVar$P.Value),]
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = ZoneVariance.original)) + 
  geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$ZoneVariance.original)

summary(summa.fit.zoneVar)
head(toptable.zoneVar, 10)
```


```{r load_results_depEn}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/DependenceEntropy.original.rda")

summa.fit.depEn <- decideTests(fit)
summary(summa.fit.depEn)
```


```{r load_results_depNU}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/DependenceNonUniformity.original.rda")

summa.fit.depNU <- decideTests(fit)
toptable.depNU <- topTable(fit, coef="DependenceNonUniformity.original", number = dim(counts.ok)[1])
toptable.depNU <- toptable.depNU[order(toptable.depNU$P.Value),]
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = DependenceNonUniformity.original)) + 
  geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$DependenceNonUniformity.original)

summary(summa.fit.depNU)
head(toptable.depNU, 10)
```


```{r load_results_depNUn}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/DependenceNonUniformityNormalized.original.rda")

summa.fit.depNUn <- decideTests(fit)
summary(summa.fit.depNUn)
```


```{r load_results_depVar}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/DependenceVariance.original.rda")

summa.fit.depVar <- decideTests(fit)
summary(summa.fit.depVar)
```


```{r load_results_hglRem}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/HighGrayLevelRunEmphasis.original.rda")

summa.fit.hglRem <- decideTests(fit)
toptable.hglRem <- topTable(fit, coef="HighGrayLevelRunEmphasis.original", number = dim(counts.ok)[1])
toptable.hglRem <- toptable.hglRem[order(toptable.hglRem$P.Value),]
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = HighGrayLevelRunEmphasis.original)) + 
  geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$HighGrayLevelRunEmphasis.original)

summary(summa.fit.hglRem)
head(toptable.hglRem, 10)
```


```{r load_results_larDepEm}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/LargeDependenceEmphasis.original.rda")

summa.fit.larDepEm <- decideTests(fit)
toptable.larDepEm <- topTable(fit, coef="LargeDependenceEmphasis.original", number = dim(counts.ok)[1])
toptable.larDepEm <- toptable.larDepEm[order(toptable.larDepEm$P.Value),]
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = LargeDependenceEmphasis.original)) + 
  geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$LargeDependenceEmphasis.original)

summary(summa.fit.larDepEm)
head(toptable.larDepEm)
```


```{r load_results_larDepHigh}
  load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/LargeDependenceHighGrayLevelEmphasis.original.rda")

summa.fit.larDepHigh <- decideTests(fit)
toptable.larDepHigh <- topTable(fit, coef="LargeDependenceHighGrayLevelEmphasis.original", number = dim(counts.ok)[1])
toptable.larDepHigh <- toptable.larDepHigh[order(toptable.larDepHigh$P.Value),]
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = LargeDependenceHighGrayLevelEmphasis.original)) + 
  geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$LargeDependenceHighGrayLevelEmphasis.original)

summary(summa.fit.larDepHigh)
head(toptable.larDepHigh, 10)
```


```{r load_results_larDepLow}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/LargeDependenceLowGrayLevelEmphasis.original.rda")

summa.fit.larDepLow <- decideTests(fit)
toptable.larDepLow <- topTable(fit, coef="LargeDependenceLowGrayLevelEmphasis.original", number = dim(counts.ok)[1])
toptable.larDepLow <- toptable.larDepLow[order(toptable.larDepLow$P.Value),]
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = LargeDependenceLowGrayLevelEmphasis.original)) + 
  geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$LargeDependenceLowGrayLevelEmphasis.original)

summary(summa.fit.larDepLow)
head(toptable.larDepLow, 10)
```


```{r load_results_smDepEm}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/SmallDependenceEmphasis.original.rda")

summa.fit.smDepEm <- decideTests(fit)
summary(summa.fit.smDepEm)
```


```{r load_results_smDepHigh}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/SmallDependenceHighGrayLevelEmphasis.original.rda")

summa.fit.smDepHigh <- decideTests(fit)
summary(summa.fit.smDepHigh)
```


```{r load_results_smDepLow}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/SmallDependenceLowGrayLevelEmphasis.original.rda")

summa.fit.smDepLow <- decideTests(fit)
summary(summa.fit.smDepLow)
```


```{r load_results_busy}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/Busyness.original.rda")

summa.fit.busy <- decideTests(fit)
toptable.busy <- topTable(fit, coef="Busyness.original", number = dim(counts.ok)[1])
toptable.busy <- toptable.busy[order(toptable.busy$P.Value),]
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = Busyness.original)) + 
  geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$Busyness.original)

summary(summa.fit.busy)
head(toptable.busy, 10)
```


```{r load_results_coar}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/Coarseness.original.rda")

summa.fit.coar <- decideTests(fit)
toptable.coar <- topTable(fit, coef="Coarseness.original", number = dim(counts.ok)[1])
toptable.coar <- toptable.coar[order(toptable.coar$P.Value),]
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = Coarseness.original)) + 
  geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$Coarseness.original)

summary(summa.fit.coar)
head(toptable.coar, 10)
```


```{r load_results_comp}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/Complexity.original.rda")

summa.fit.comp <- decideTests(fit)
summary(summa.fit.comp)
```


```{r load_results_strength}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/Strength.original.rda")

summa.fit.strength <- decideTests(fit)
toptable.strength <- topTable(fit, coef="Strength.original", number = dim(counts.ok)[1])
toptable.strength <- toptable.strength[order(toptable.strength$P.Value),]
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = Strength.original)) + 
  geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$Strength.original)

summary(summa.fit.strength)
head(toptable.strength, 10)
```


```{r load_results_imc1}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/Imc1.original.rda")
summa.fit.imc1 <- decideTests(fit)
summary(summa.fit.imc1)
```


```{r load_results_imc2}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/Imc2.original.rda")
summa.fit.imc2 <- decideTests(fit)
summary(summa.fit.imc2)
```


```{r load_results_X10}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/X10Percentile.original.rda")
summa.fit.X10 <- decideTests(fit)
toptable.X10 <- topTable(fit, coef="X10Percentile.original", number = dim(counts.ok)[1])
toptable.X10 <- toptable.X10[order(toptable.X10$P.Value),]
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = X10Percentile.original)) + 
  geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$X10Percentile.original)

summary(summa.fit.X10)
head(toptable.X10, 10)
```


```{r load_results_X90}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/X90Percentile.original.rda")
summa.fit.X90 <- decideTests(fit)
summary(summa.fit.X90)
```


```{r load_results_ent}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/Entropy.original.rda")
summa.fit.ent <- decideTests(fit)
summary(summa.fit.ent)
```


```{r load_results_engy}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/Energy.original.rda")
summa.fit.engy <- decideTests(fit)
summary(summa.fit.engy)
```


```{r load_results_intq}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/InterquartileRange.original.rda")

summa.fit.intq <- decideTests(fit)
toptable.intq <- topTable(fit, coef="InterquartileRange.original", number = dim(counts.ok)[1])
toptable.intq <- toptable.intq[order(toptable.intq$P.Value),]
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = InterquartileRange.original)) + 
  geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$InterquartileRange.original)

summary(summa.fit.intq)
head(toptable.intq, 10)
```


```{r load_results_kur}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/Kurtosis.original.rda")
summa.fit.kur <- decideTests(fit)
summary(summa.fit.kur)
```


```{r load_results_max}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/Maximum.original.rda")
summa.fit.max <- decideTests(fit)
summary(summa.fit.max)
```


```{r load_results_meanAb}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/MeanAbsoluteDeviation.original.rda")
summa.fit.meanAb <- decideTests(fit)
summary(summa.fit.meanAb)
```


```{r load_results_mean}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/Mean.original.rda")
summa.fit.mean <- decideTests(fit)
summary(summa.fit.mean)
```


```{r load_results_med}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/Median.original.rda")
summa.fit.med <- decideTests(fit)
summary(summa.fit.med)
```


```{r load_results_min}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/Minimum.original.rda")

summa.fit.min <- decideTests(fit)
toptable.min <- topTable(fit, coef="Minimum.original", number = dim(counts.ok)[1])
toptable.min <- toptable.min[order(toptable.min$P.Value),]
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = Minimum.original)) + 
  geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$Minimum.original)

summary(summa.fit.min)
head(toptable.min, 10)
```


```{r load_results_range}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/Range.original.rda")

summa.fit.range <- decideTests(fit)
toptable.range <- topTable(fit, coef="Range.original", number = dim(counts.ok)[1])
toptable.range <- toptable.range[order(toptable.range$P.Value),]
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = Range.original)) + 
  geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$Range.original)

summary(summa.fit.range)
head(toptable.range, 10)
```


```{r load_results_robustM}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/RobustMeanAbsoluteDeviation.original.rda")

summa.fit.robustM <- decideTests(fit)
toptable.robustM <- topTable(fit, coef="RobustMeanAbsoluteDeviation.original", number = dim(counts.ok)[1])
toptable.robustM <- toptable.robustM[order(toptable.robustM$P.Value),]
p.val.voom <- as.data.frame(fit$p.value)

# P-value distribution of results computed by limma
ggplot(data = p.val.voom, aes(x = RobustMeanAbsoluteDeviation.original)) + 
  geom_histogram(bins = 100)

# QQplot plot for p-values computed by limma
GWASTools::qqPlot(p.val.voom$RobustMeanAbsoluteDeviation.original)

summary(summa.fit.robustM)
head(toptable.robustM, 10)
```


```{r load_results_root}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/RootMeanSquared.original.rda")
summa.fit.root <- decideTests(fit)
summary(summa.fit.root)
```


```{r load_results_skw}
load("/Users/carlacasanovasuarez/Desktop/Radiomic features models (indiv)/Skewness.original.rda")
summa.fit.skw <- decideTests(fit)
summary(summa.fit.skw)
```


## Enrichment    

Load RadAR object in order to retrieve `rowdata` later:  

```{r load_rdr}
load("/Users/carlacasanovasuarez/Documents/Master Bioinformatics UAB/Prácticas Radiomics/Radiomic features/Results_rfeatures/radar_L1_Norm_scaled_adjusted_combat_final.rda")

rdr_L1_final
```

Load annotation data of genes stored in `annotation.sputum.ok` object to check which ones have annotations:  

```{r load_anno}
load("/Users/carlacasanovasuarez/Documents/Master Bioinformatics UAB/Prácticas Radiomics/Radiomic features/Results_rfeatures/R objects/sputum_eset_annotationOK.rda")
```

Create a vector with feature names of features positively associated with genes DE:  

```{r positive_features_Combat}
pos_rf <- list("MeshVolume.original",
               "Sphericity.original",
               "VoxelVolume.original",
               "Variance.original",
               "Autocorrelation.original",
               "ClusterShade.original",
               "JointAverage.original",
               "SumAverage.original",
               "HighGrayLevelEmphasis.original",
               "LongRunLowGrayLevelEmphasis.original",
               "HighGrayLevelZoneEmphasis.original",
               "LargeAreaEmphasis.original",
               "LowGrayLevelEmphasis.original",
               "LargeAreaHighGrayLevelEmphasis.original",
               "LargeAreaLowGrayLevelEmphasis.original",
               "LowGrayLevelZoneEmphasis.original",
               "SmallAreaEmphasis.original",
               "SmallAreaHighGrayLevelEmphasis.original",
               "SmallAreaLowGrayLevelEmphasis.original",
               "ZoneVariance.original",
               "DependenceNonUniformity.original",
               "LargeDependenceHighGrayLevelEmphasis.original",
               "LargeDependenceLowGrayLevelEmphasis.original",
               "Busyness.original",
               "Coarseness.original",
               "Strength.original",
               "InterquartileRange.original",
               "Range.original")
```


Create a data frame with information about **feature type** and **clustering** of radiomic features associated to DE genes:  

```{r check_feature_type}
set.seed(1234)

# Change feature name to match rdr row data (crop ".original")
feature_names <- substr(pos_rf, 1, nchar(pos_rf)-9) 

# Check order
identical(rowData(rdr_L1_final)$feature_name, feature_names)
feature_names_ordered <- intersect(rowData(rdr_L1_final)$feature_name, feature_names)

# Check row data of rdr object: feature name and feature type
ids <- which(rowData(rdr_L1_final)$feature_name %in% feature_names_ordered)
identical(rowData(rdr_L1_final)[ids, "feature_name"], feature_names_ordered)
out <- cbind(feature_names_ordered,
              feature.type = rowData(rdr_L1_final)[ids, "feature_type"])
out <- as.data.frame(out)

# Compute clusters and add them to the final dataframe
rdr_positive <- rdr_assay[unlist(pos_rf),]
k.res <- kmeans(rdr_positive, centers = 6)

# Check order of kmeans clustering results and out dataframe. CHANGE IT!
identical(substr(rownames(rdr_positive), 1, nchar(rownames(rdr_positive))-9), out$feature_names_ordered)
k.results.order <- intersect(substr(names(k.res$cluster), 1, nchar(names(k.res$cluster))-9), out$feature_names_ordered)
rownames(out) <- out$feature_names_ordered
out <- out[k.results.order,]

# Add results from k.means properly ordered 
out$cluster <- k.res$cluster
out <- out %>% arrange(cluster, feature.type)

out
```


### Cluster 1  
Check features in *cluster 1*:  

```{r fo_subset}
cluster1.DE <- out[out$cluster == 1,]

# In this case, the only features correlated are LowGrayLevelEmphasis and Autocorrelation
cluster1.DE
```

Store names of genes DE (p.adj < 0.01 if possible):  

```{r de_genes_1}
# Create bools for genes with the most restrictive p.value 
dd.1.Variance <- toptable.variance$adj.P.Val < 0.05 & !is.na(toptable.variance$adj.P.Val)
dd.1.clustSh <- toptable.clustSh$adj.P.Val < 0.05 & !is.na(toptable.clustSh$adj.P.Val)
dd.1.hglZEM <- toptable.hglZEM$adj.P.Val < 0.05 & !is.na(toptable.hglZEM$adj.P.Val)
dd.1.lowGLem <- toptable.lowGLem$adj.P.Val < 0.05 & !is.na(toptable.lowGLem$adj.P.Val)
dd.1.smAHem <- toptable.smAHem$adj.P.Val < 0.05 & !is.na(toptable.smAHem$adj.P.Val)
dd.1.larDepLow <- toptable.larDepLow$adj.P.Val < 0.05 & !is.na(toptable.larDepLow$adj.P.Val)
dd.1.busy <- toptable.busy$adj.P.Val < 0.05 & !is.na(toptable.busy$adj.P.Val)
dd.1.intq <- toptable.intq$adj.P.Val < 0.05 & !is.na(toptable.intq$adj.P.Val)
dd.1.range <- toptable.range$adj.P.Val < 0.05 & !is.na(toptable.range$adj.P.Val)

deGenes.1 <- c(rownames(toptable.variance[dd.1.Variance, ]),rownames(toptable.clustSh[dd.1.clustSh, ]), rownames(toptable.hglZEM[dd.1.hglZEM, ]), rownames(toptable.lowGLem[dd.1.lowGLem,]), rownames(toptable.smAHem[dd.1.smAHem,]), rownames(toptable.larDepLow[dd.1.larDepLow,]), rownames(toptable.busy[dd.1.busy,]), rownames(toptable.intq[dd.1.intq,]), rownames(toptable.range[dd.1.range,]))

# Length and duplicates
length(deGenes.1)
table(duplicated(deGenes.1))

# Use unique IDs
deGenes.1 <- unique(deGenes.1)
length(deGenes.1)

deGenes.1

#save(deGenes.1, file = "list DE genes cluster 1.rda")
#load("list DE genes cluster 1.rda")
```

Check if there are annotations available:  

```{r de_genes_ann_1}
# Check info 
annotation.sputum.ok[deGenes.1,]
```

Create gene universe:  

```{r gene_universe_1}
geneUniverse.1 <- rownames(toptable.variance[!is.na(toptable.variance$P.Value), ]) 
length(geneUniverse.1)
```

Change annotation to `ENTREZ`:  

```{r entrez_annotation_1}
# Retrieve ENTREZ names 
deGenes.1.entrez <- unlist(mget(deGenes.1, envir=hgu133plus2ENTREZID, ifnotfound = NA))
#geneUniverse.1.entrez <- unlist(mget(geneUniverse.1, envir=hgu133plus2ENTREZID, ifnotfound = NA))
```

**GO** enrichment:  

```{r GO_enrichment_1}
ans.go.1 <- enrichGO(gene = deGenes.1.entrez, ont = "BP",
                     #universe = geneUniverse.1.entrez,
                     OrgDb ="org.Hs.eg.db", 
                     readable=TRUE, pvalueCutoff = 1,
                     qvalueCutoff = 1)

# See results
tab.go.1 <- as.data.frame(ans.go.1@result) 
#tab.go.1 <- subset(tab.go.1, Count>2) 
tab.go.1[1:5, 1:6]
```

**KEEG** enrichment:  

```{r kegg_1}
ans.kegg.1 <- enrichKEGG(gene = deGenes.1.entrez,
                          #universe = geneUniverse.1.entrez,
                         organism = 'hsa',
                         pvalueCutoff = 1,
                         qvalueCutoff = 1) 

# See results
tab.kegg.1 <- as.data.frame(ans.kegg.1@result)
#tab.kegg.1 <- subset(tab.kegg.1, Count>2) 
tab.kegg.1[1:5, 1:6]
```

**DisGeNet** analysis. Get API:  

```{r disgenet_get_API}
library(disgenet2r)

# Get API
disgenet_api_key <- get_disgenet_api_key(email = "carlacasanovasuarez@gmail.com", password = "mypassword1234", verbose = TRUE)

Sys.setenv(DISGENET_API_KEY= disgenet_api_key)
```


```{r disgenet_enrichment_1}
res_enrich.1 <-disease_enrichment(entities = deGenes.1.entrez, vocabulary = "ENTREZ", database = "CURATED")

res.table.disgenet.1 <- res_enrich.1@qresult
res.table.disgenet.1

#save(res.table.disgenet.1, file = "/Users/carlacasanovasuarez/Documents/R directory/Radiomic features/Transcriptomic analysis/DisGenet_1_features.rda")
```

**MSigDb** enrichment analysis (group C7 inmunologic signature):  

```{r msigdb_1}
# Load terms for genes coded by ENTREZ of the gene set C7
c3.tf <- read.gmt("/Users/carlacasanovasuarez/Downloads/c7.all.v7.5.1.entrez.gmt.txt")

# Enrichment
ans.tf.1 <- enricher(deGenes.1.entrez, TERM2GENE=c3.tf, pvalueCutoff =1) 
tab.tf.1 <- as.data.frame(ans.tf.1@result)
#tab.tf.f1<- subset(tab.tf.f1, Count>5) 
tab.tf.1[1:5,1:5]
```


### Cluster 2    

```{r clust2_subset}
cluster2.DE <- out[out$cluster == 2,]

# In this case, these features are not correlated
cluster2.DE
```

Store names of genes DE (p.adj < 0.01 if possible):  

```{r clust2_genes}
# Create bools for genes with the most restrictive p.value 
dd.2.MeshVol <- toptable.MeshVol$adj.P.Val < 0.05 & !is.na(toptable.MeshVol$adj.P.Val)
dd.2.voxVol <- toptable.voxVol$adj.P.Val < 0.05 & !is.na(toptable.voxVol$adj.P.Val)
dd.2.depNU <- toptable.depNU$adj.P.Val < 0.05 & !is.na(toptable.depNU$adj.P.Val)

deGenes.2 <- c(rownames(toptable.MeshVol[dd.2.MeshVol, ]),rownames(toptable.voxVol[dd.2.voxVol, ]), rownames(toptable.depNU[dd.2.depNU,]))

# Check duplicates
length(deGenes.2)
table(duplicated(deGenes.2))

# Remove duplicates
deGenes.2 <- unique(deGenes.2)

deGenes.2
```

Check if there are annotations available:  

```{r de_genes_ann_clust2}
# Check info 
annotation.sputum.ok[deGenes.2,]
```

Gene universe shape:  

```{r gene_universe_sh, eval=FALSE}
geneUniverse.sh <- rownames(toptable.sphericity[!is.na(toptable.sphericity$P.Value), ]) 
length(geneUniverse.sh)
```

Change annotation to `ENTREZ`:  

```{r entrez_annotation_clust2}
# Retrieve ENTREZ names 
deGenes.2.entrez <- unlist(mget(deGenes.2, envir=hgu133plus2ENTREZID, ifnotfound = NA))
#geneUniverse.sh.entrez <- unlist(mget(geneUniverse.sh, envir=hgu133plus2ENTREZID, ifnotfound = NA))
```

**GO** enrichment:  

```{r GO_enrichment_clust2}
ans.go.2 <- enrichGO(gene = deGenes.2.entrez, ont = "BP",
                      #universe = geneUniverse.sh.entrez,
                      OrgDb ="org.Hs.eg.db", 
                      readable=TRUE, pvalueCutoff = 0.05)

# See results
tab.go.2 <- as.data.frame(ans.go.2@result) 
#tab.go.2<- subset(tab.go.2, Count>5) 
tab.go.2[1:5, 1:6]
```

**KEGG** enrichment:  

```{r kegg_clust2, eval=FALSE}
ans.kegg.2 <- enrichKEGG(gene = deGenes.2.entrez, 
                       organism = 'hsa',
                       #universe = geneUniverse.sh.entrez,
                       pvalueCutoff = 1,
                       qvalueCutoff = 1) 

# See results
tab.kegg.2 <- as.data.frame(ans.kegg.2@result)
#tab.kegg.2 <- subset(tab.kegg.2, Count>5) 
tab.kegg.2[1:5, 1:6]
```

Disgenet:  

```{r disgenet_enrichment_clust2}
res_enrich.2 <-disease_enrichment(entities = deGenes.2.entrez, vocabulary = "ENTREZ", database = "CURATED")

res.table.disgenet.2 <- res_enrich.2@qresult
res.table.disgenet.2

#save(res.table.disgenet.2, file = "/Users/carlacasanovasuarez/Documents/R directory/Radiomic features/Transcriptomic analysis/DisGenet_2_features.rda")
```

**MSigDb** enrichment analysis (group C7 inmunologic signature):  

```{r msigdb_clust2}
# Load terms for genes coded by ENTREZ of the gene set C7
c3.tf <- read.gmt("/Users/carlacasanovasuarez/Downloads/c7.all.v7.5.1.entrez.gmt.txt")

# Enrichment
ans.tf.2 <- enricher(deGenes.2.entrez, TERM2GENE=c3.tf) 
tab.tf.2 <- as.data.frame(ans.tf.2@result)
#tab.tf.2 <- subset(tab.tf.2, Count>5) 
tab.tf.2[1:5,1:5]
```


### cluster 3  

```{r clust3_subset}
cluster3.DE <- out[out$cluster == 3,]

cluster3.DE
```

Store names of genes DE (p.adj < 0.01 if possible):  

```{r clust3_genes}
# Create bools for genes with the most restrictive p.value 
dd.3.zoneVar <- toptable.zoneVar$adj.P.Val < 0.05 & !is.na(toptable.zoneVar$adj.P.Val)
dd.3.strength <- toptable.strength$adj.P.Val < 0.05 & !is.na(toptable.strength$adj.P.Val) 

deGenes.3 <- c(rownames(toptable.zoneVar[dd.3.zoneVar, ]),rownames(toptable.strength[dd.3.strength, ])) 

# Check duplicated genes
length(deGenes.3)
table(duplicated(deGenes.3))

# Filter duplicates
deGenes.3 <- unique(deGenes.3)

deGenes.3
```

Check if there are annotations available:  

```{r de_clust3_ann}
# Check info 
annotation.sputum.ok[deGenes.3,]
```

Change annotation to `ENTREZ`:  

```{r entrez_annotation_clust3}
# Retrieve ENTREZ names 
deGenes.3.entrez <- unlist(mget(deGenes.3, envir=hgu133plus2ENTREZID, ifnotfound = NA))
#geneUniverse.glcm.entrez <- unlist(mget(geneUniverse.glcm, envir=hgu133plus2ENTREZID, ifnotfound = NA))
```

**GO** enrichment:  

```{r GO_enrichment_clust3}
ans.go.3 <- enrichGO(gene = deGenes.3.entrez, ont = "BP",
                        #universe = geneUniverse.glcm.entrez,
                     OrgDb ="org.Hs.eg.db", 
                     readable=TRUE,
                     pvalueCutoff = 1,
                     qvalueCutoff = 1) 

# See results
tab.go.3 <- as.data.frame(ans.go.3@result) 
#tab.go.3 <- subset(tab.go.3, Count>5) 
tab.go.3[1:5, 1:6]
```

**KEGG** enrichment:  

```{r kegg_clust3}
ans.kegg.3 <- enrichKEGG(gene = deGenes.3.entrez, 
                            #universe = geneUniverse.glcm.entrez,
                         organism = 'hsa',
                         pvalueCutoff = 1,
                         qvalueCutoff = 1) 

# See results
tab.kegg.3 <- as.data.frame(ans.kegg.3@result)
#tab.kegg.3 <- subset(tab.kegg.3, Count>5) 
tab.kegg.3[1:5, 1:6]
```

Disgenet:  

```{r disgenet_enrichment_clust3}
res_enrich.3 <-disease_enrichment(entities = deGenes.3.entrez, vocabulary = "ENTREZ", database = "CURATED")

res.table.disgenet.3 <- res_enrich.3@qresult
res.table.disgenet.3

#save(res.table.disgenet.3, file = "/Users/carlacasanovasuarez/Documents/R directory/Radiomic features/Transcriptomic analysis/DisGenet_3_features.rda")
```

**MSigDb** enrichment analysis (group C7 inmunologic signature):  

```{r msigdb_clust3}
# Load terms for genes coded by ENTREZ of the gene set C7
c3.tf <- read.gmt("/Users/carlacasanovasuarez/Downloads/c7.all.v7.5.1.entrez.gmt.txt")

# Enrichment
ans.tf.3 <- enricher(deGenes.3.entrez, TERM2GENE=c3.tf, pvalueCutoff =1) 
tab.tf.3 <- as.data.frame(ans.tf.3@result)
#tab.tf.glcm <- subset(tab.tf.glcm, Count>5) 
tab.tf.3[1:5,1:5]
```


### Cluster 4    

```{r clust4_subset}
cluster4.DE <- out[out$cluster == 4,]

cluster4.DE
```

Store names of genes DE (p.adj < 0.01 if possible):  

```{r clust4_genes}
# Create bools for genes with the most restrictive p.value 
dd.4.sphe <- toptable.sphericity$adj.P.Val < 0.05 & !is.na(toptable.sphericity$adj.P.Val)

deGenes.4 <- c(rownames(toptable.sphericity[dd.4.sphe, ]))

# Check duplicates
length(deGenes.4)
table(duplicated(deGenes.4))

# Remove duplicates
deGenes.4 <- unique(deGenes.4)

deGenes.4
```

Check if there are annotations available:  

```{r de_genes_ann_clust4}
# Check info 
annotation.sputum.ok[deGenes.4,]
```

Change annotation to `ENTREZ`:  

```{r entrez_annotation_clust4}
# Retrieve ENTREZ names 
deGenes.4.entrez <- unlist(mget(deGenes.4, envir=hgu133plus2ENTREZID, ifnotfound = NA))
#geneUniverse.glcm.entrez <- unlist(mget(geneUniverse.glcm, envir=hgu133plus2ENTREZID, ifnotfound = NA))
```

**GO** enrichment:  

```{r GO_enrichment_clust4}
ans.go.4 <- enrichGO(gene = deGenes.4.entrez, ont = "BP",
                        #universe = geneUniverse.glcm.entrez,
                     OrgDb ="org.Hs.eg.db", 
                     readable=TRUE,
                     pvalueCutoff = 1,
                     qvalueCutoff = 1) 

# See results
tab.go.4 <- as.data.frame(ans.go.4@result) 
#tab.go.3 <- subset(tab.go.3, Count>5) 
tab.go.4[1:5, 1:6]
```

**KEGG** enrichment:  

```{r kegg_clust4}
ans.kegg.4 <- enrichKEGG(gene = deGenes.4.entrez, 
                            #universe = geneUniverse.glcm.entrez,
                         organism = 'hsa',
                         pvalueCutoff = 1,
                         qvalueCutoff = 1) 

# See results
tab.kegg.4 <- as.data.frame(ans.kegg.4@result)
#tab.kegg.3 <- subset(tab.kegg.3, Count>5) 
tab.kegg.4[1:5, 1:6]
```

Disgenet:  

```{r disgenet_enrichment_clust4}
res_enrich.4 <-disease_enrichment(entities = deGenes.4.entrez, vocabulary = "ENTREZ", database = "CURATED")

res.table.disgenet.4 <- res_enrich.4@qresult
res.table.disgenet.4

#save(res.table.disgenet.4, file = "/Users/carlacasanovasuarez/Documents/R directory/Radiomic features/Transcriptomic analysis/DisGenet_4_features.rda")
```

**MSigDb** enrichment analysis (group C7 inmunologic signature):  

```{r msigdb_clust4}
# Load terms for genes coded by ENTREZ of the gene set C7
c3.tf <- read.gmt("/Users/carlacasanovasuarez/Downloads/c7.all.v7.5.1.entrez.gmt.txt")

# Enrichment
ans.tf.4 <- enricher(deGenes.4.entrez, TERM2GENE=c3.tf, pvalueCutoff =1) 
tab.tf.4 <- as.data.frame(ans.tf.4@result)
#tab.tf.glcm <- subset(tab.tf.glcm, Count>5) 
tab.tf.4[1:5,1:5]
```


### Cluster 5  

```{r clust5_subset}
cluster5.DE <- out[out$cluster == 5,]

cluster5.DE
```

Store names of genes DE (p.adj < 0.01 if possible):  

```{r clust5_genes}
# Create bools for genes with the most restrictive p.value 
dd.5.autoc <- toptable.autoc$adj.P.Val < 0.05 & !is.na(toptable.autoc$adj.P.Val)
dd.5.jAv <- toptable.jAv$adj.P.Val < 0.05 & !is.na(toptable.jAv$adj.P.Val)
dd.5.sumAv <- toptable.sumAv$adj.P.Val < 0.05 & !is.na(toptable.sumAv$adj.P.Val)
dd.5.hglEm <- toptable.hglEm$adj.P.Val < 0.05 & !is.na(toptable.hglEm$adj.P.Val)
dd.5.lowGLZem <- toptable.lowGLZem$adj.P.Val < 0.05 & !is.na(toptable.lowGLZem$adj.P.Val)
dd.5.smAem <- toptable.smAem$adj.P.Val < 0.05 & !is.na(toptable.smAem$adj.P.Val)
dd.5.smALem <- toptable.smALem$adj.P.Val < 0.05 & !is.na(toptable.smALem$adj.P.Val)
dd.5.larDepHig <- toptable.larDepHigh$adj.P.Val < 0.05 & !is.na(toptable.larDepHigh$adj.P.Val)

# Store names of DE genes
deGenes.5 <- c(rownames(toptable.autoc[dd.5.autoc, ]),rownames(toptable.jAv[dd.5.jAv, ]), rownames(toptable.sumAv[dd.5.sumAv, ]), rownames(toptable.hglEm[dd.5.hglEm,]), rownames(toptable.lowGLZem[dd.5.lowGLZem,]), rownames(toptable.smAem[dd.5.smAem,]), rownames(toptable.smALem[dd.5.smALem,]), rownames(toptable.larDepHigh[dd.5.larDepHig,]))

# Check duplicates
length(deGenes.5)
table(duplicated(deGenes.5))

# Remove duplicates
deGenes.5 <- unique(deGenes.5)

# Check if some of these genes had been already associated with other cluster
length(deGenes.5[which(deGenes.5 %in% deGenes.1)])
length(deGenes.5[which(deGenes.5 %in% deGenes.2)])
length(deGenes.5[which(deGenes.5 %in% deGenes.3)])
length(deGenes.5[which(deGenes.5 %in% deGenes.4)])
#repeated_genes <- deGenes.5[which(deGenes.5 %in% deGenes.1)]
#deGenes.5 <- deGenes.5[which(!(deGenes.5 %in% repeated_genes))]

deGenes.5
```

Check if there are annotations available:  

```{r de_genes_ann_clust5}
# Check info 
annotation.sputum.ok[deGenes.5,]
```

Change annotation to `ENTREZ`:  

```{r entrez_annotation_clust5}
# Retrieve ENTREZ names 
deGenes.5.entrez <- unlist(mget(deGenes.5, envir=hgu133plus2ENTREZID, ifnotfound = NA))
#geneUniverse.glcm.entrez <- unlist(mget(geneUniverse.glcm, envir=hgu133plus2ENTREZID, ifnotfound = NA))
```

**GO** enrichment:  

```{r GO_enrichment_clust5}
ans.go.5 <- enrichGO(gene = deGenes.5.entrez, ont = "BP",
                        #universe = geneUniverse.glcm.entrez,
                     OrgDb ="org.Hs.eg.db", 
                     readable=TRUE,
                     pvalueCutoff = 1,
                     qvalueCutoff = 1) 

# See results
tab.go.5 <- as.data.frame(ans.go.5@result) 
#tab.go.5 <- subset(tab.go.5, Count>5)
tab.go.5[1:5, 1:6]
```

**KEGG** enrichment:  

```{r kegg_clust5}
ans.kegg.5 <- enrichKEGG(gene = deGenes.5.entrez, 
                            #universe = geneUniverse.glcm.entrez,
                         organism = 'hsa',
                         pvalueCutoff = 1,
                         qvalueCutoff = 1) 

# See results
tab.kegg.5 <- as.data.frame(ans.kegg.5@result)
#tab.kegg.3 <- subset(tab.kegg.3, Count>5) 
tab.kegg.5[1:5, 1:6]
```

Disgenet:  

```{r disgenet_enrichment_clust5}
res_enrich.5 <-disease_enrichment(entities = deGenes.5.entrez, vocabulary = "ENTREZ", database = "CURATED")

res.table.disgenet.5 <- res_enrich.5@qresult
res.table.disgenet.5

#save(res.table.disgenet.5, file = "/Users/carlacasanovasuarez/Documents/R directory/Radiomic features/Transcriptomic analysis/DisGenet_5_features.rda")
```

**MSigDb** enrichment analysis (group C7 inmunologic signature):  

```{r msigdb_clust5}
# Load terms for genes coded by ENTREZ of the gene set C7
c3.tf <- read.gmt("/Users/carlacasanovasuarez/Downloads/c7.all.v7.5.1.entrez.gmt.txt")

# Enrichment
ans.tf.5 <- enricher(deGenes.5.entrez, TERM2GENE=c3.tf) 
tab.tf.5 <- as.data.frame(ans.tf.5@result)
#tab.tf.glcm <- subset(tab.tf.glcm, Count>5) 
tab.tf.5[1:5,1:5]
```


### Cluster 6  

```{r clust6_subset}
cluster6.DE <- out[out$cluster == 6,]

cluster6.DE
```

Store names of genes DE (p.adj < 0.01 if possible):  

```{r clust6_genes}
# Create bools for genes with the most restrictive p.value 
dd.6.lrLEm <- toptable.lrLEm$adj.P.Val < 0.05 & !is.na(toptable.lrLEm$adj.P.Val)
dd.6.larAEm <- toptable.larAEm$adj.P.Val < 0.05 & !is.na(toptable.larAEm$adj.P.Val)
dd.6.larAHigh <- toptable.larAHigh$adj.P.Val < 0.05 & !is.na(toptable.larAHigh$adj.P.Val)
dd.6.larALow <- toptable.larALow$adj.P.Val < 0.05 & !is.na(toptable.larALow$adj.P.Val)
dd.6.coar <- toptable.coar$adj.P.Val < 0.05 & !is.na(toptable.coar$adj.P.Val)



# Store names of DE genes
deGenes.6 <- c(rownames(toptable.lrLEm[dd.6.lrLEm, ]),rownames(toptable.larAEm[dd.6.larAEm, ]), rownames(toptable.larAHigh[dd.6.larAHigh, ]), rownames(toptable.larALow[dd.6.larALow,]), rownames(toptable.coar[dd.6.coar,]))

# Check duplicates
length(deGenes.6)
table(duplicated(deGenes.6))

# Remove duplicates
deGenes.6 <- unique(deGenes.6)

# Check if some of these genes had been already associated with other cluster
length(deGenes.6[which(deGenes.6 %in% deGenes.1)])
length(deGenes.6[which(deGenes.6 %in% deGenes.2)])
length(deGenes.6[which(deGenes.6 %in% deGenes.3)])
length(deGenes.6[which(deGenes.6 %in% deGenes.4)])
length(deGenes.6[which(deGenes.6 %in% deGenes.5)])
#repeated_genes <- deGenes.6[which(deGenes.6 %in% deGenes.5 | deGenes.6 %in% deGenes.1)]
#deGenes.6 <- deGenes.6[which(!(deGenes.6 %in% repeated_genes))]

deGenes.6
```

Check if there are annotations available:  

```{r de_genes_ann_clust6}
# Check info 
annotation.sputum.ok[deGenes.6,]
```

Change annotation to `ENTREZ`:  

```{r entrez_annotation_clust6}
# Retrieve ENTREZ names 
deGenes.6.entrez <- unlist(mget(deGenes.6, envir=hgu133plus2ENTREZID, ifnotfound = NA))
#geneUniverse.glcm.entrez <- unlist(mget(geneUniverse.glcm, envir=hgu133plus2ENTREZID, ifnotfound = NA))
```

**GO** enrichment:  

```{r GO_enrichment_clust6}
ans.go.6 <- enrichGO(gene = deGenes.6.entrez, ont = "BP",
                        #universe = geneUniverse.glcm.entrez,
                     OrgDb ="org.Hs.eg.db", 
                     readable=TRUE,
                     pvalueCutoff = 1,
                     qvalueCutoff = 1) 

# See results
tab.go.6 <- as.data.frame(ans.go.6@result) 
#tab.go.3 <- subset(tab.go.3, Count>5) 
tab.go.6[1:5, 1:6]
```


**KEGG** enrichment:  

```{r kegg_clust6}
ans.kegg.6 <- enrichKEGG(gene = deGenes.6.entrez, 
                            #universe = geneUniverse.glcm.entrez,
                         organism = 'hsa',
                         pvalueCutoff = 1,
                         qvalueCutoff = 1) 

# See results
tab.kegg.6 <- as.data.frame(ans.kegg.6@result)
#tab.kegg.3 <- subset(tab.kegg.3, Count>5) 
tab.kegg.6[1:5, 1:6]
```

Disgenet:  

```{r disgenet_enrichment_clust6}
res_enrich.6 <-disease_enrichment(entities = deGenes.6.entrez, vocabulary = "ENTREZ", database = "CURATED")

res.table.disgenet.6 <- res_enrich.6@qresult
res.table.disgenet.6

#save(res.table.disgenet.5, file = "/Users/carlacasanovasuarez/Documents/R directory/Radiomic features/Transcriptomic analysis/DisGenet_5_features.rda")
```

**MSigDb** enrichment analysis (group C7 inmunologic signature):  

```{r msigdb_clust6}
# Load terms for genes coded by ENTREZ of the gene set C7
c3.tf <- read.gmt("/Users/carlacasanovasuarez/Downloads/c7.all.v7.5.1.entrez.gmt.txt")

# Enrichment
ans.tf.6 <- enricher(deGenes.6.entrez, TERM2GENE=c3.tf) 
tab.tf.6 <- as.data.frame(ans.tf.6@result)
#tab.tf.glcm <- subset(tab.tf.glcm, Count>5) 
tab.tf.6[1:5,1:5]
```

## Summary  

Cluster 1:  

```{r 1_plots}
# GO
pclus1GO <- dotplot(ans.go.1, showCategory=10) + ggtitle("GO of DE genes from sputum associated with Cluster 1") 

# KEGG
x1 <- pairwise_termsim(ans.kegg.1)
pclus1KEEG <- emapplot(x1, showCategory = 10) + ggtitle("KEEG of DE genes from sputum associated with Cluster 1") 

# Disgenet plot() function only displays diseases from ratios > 0.4. 
pclus1Dis <- plot(res_enrich.1, class = "Enrichment",  cutoff= 1) + ggtitle("DisGeNET (sputum) with Cluster 1") 

# MSigDb immune signatures
#pclus1Msig <- dotplot(ans.tf.1) + ggtitle("MSigDb immunologic signatures")

pAllclus1 <- plot_grid(pclus1GO, pclus1Dis, pclus1KEEG, labels = c("A", "B", "C"))

ggsave("Enrichment Cluster 1 NEW.pdf",pAllclus1, width=7, height=7, units="cm", scale=5)
```


Cluster 2:  

```{r 2_plots}
# GO
pclus2GO <- dotplot(ans.go.2, showCategory=10) + ggtitle("GO of DE genes from sputum associated with Cluster 2") 

# KEGG
#x2 <- pairwise_termsim(ans.kegg.2)
#pclus2KEEG <- emapplot(x2, showCategory = 10) + ggtitle("KEEG Cluster 2 sputum") 

# Disgenet plot() function only displays diseases from ratios > 0.4. 
#pclus2Dis <- plot(res_enrich.2, class = "Enrichment",  cutoff= 1) + ggtitle("DisGeNET Cluster 2 sputum") 

#pAllclus2 <- plot_grid(pclus2GO, pclus2Dis, labels = c("A", "B"))

# MSigDb immune signatures
pclus2Msig <- dotplot(ans.tf.2) + ggtitle("MSigDb immunologic signatures")

pAllclus2 <- plot_grid(pclus2GO, pclus2Msig, labels = c("A", "B"))

ggsave("Enrichment Cluster 2 NEW.pdf",pAllclus2, width=9, height=6, units="cm", scale=5)
```

Cluster 3:  

```{r 3_plots}
# GO
pclus3GO <- dotplot(ans.go.3, showCategory=10) + ggtitle("GO of DE genes from sputum associated with Cluster 3") 

# KEGG
x3 <- pairwise_termsim(ans.kegg.3)
pclus3KEEG <- emapplot(x3, showCategory = 10) + ggtitle("KEEG of DE genes from sputum associated with Cluster 3") 

# Disgenet plot() function only displays diseases from ratios > 0.4. 
pclus3Dis <- plot(res_enrich.3, class = "Enrichment",  cutoff= 1) + ggtitle("DisGeNET (sputum) with Cluster 3") 

pAllclus3 <- plot_grid(pclus3GO, pclus3Dis, pclus3KEEG, labels = c("A", "B", "C"))

# MSigDb immune signatures
#pclus3Msig <- dotplot(ans.tf.3) + ggtitle("MSigDb immunologic signatures")

ggsave("Enrichment Cluster 3 NEW.pdf",pAllclus3, width=7, height=7, units="cm", scale=5.5)
```

Cluster 4:  

```{r 4_plots}
# GO
pclus4GO <- dotplot(ans.go.4, showCategory=10) + ggtitle("GO of DE genes from sputum associated with Cluster 4") 

# KEGG
x4 <- pairwise_termsim(ans.kegg.4)
pclus4KEEG <- emapplot(x4, showCategory = 10) + ggtitle("KEEG of DE genes from sputum associated with Cluster 4") 

# Disgenet plot() function only displays diseases from ratios > 0.4. 
pclus4Dis <- plot(res_enrich.4, class = "Enrichment",  cutoff= 1) + ggtitle("DisGeNET (sputum) with Cluster 4") 

# MSigDb immune signatures
#pclus4Msig <- dotplot(ans.tf.4) + ggtitle("MSigDb immunologic signatures")

pAllclus4 <- plot_grid(pclus4GO, pclus4Dis, pclus4KEEG, labels = c("A", "B", "C"))

ggsave("Enrichment Cluster 4 NEW.pdf",pAllclus4, width=7, height=7, units="cm", scale=5)
```

Cluster 5:  

```{r 5_plots}
# GO
pclus5GO <- dotplot(ans.go.5, showCategory=10) + ggtitle("GO of DE genes from sputum with Cluster 5") 

# KEGG
x5 <- pairwise_termsim(ans.kegg.5)
pclus5KEEG <- emapplot(x5, showCategory = 10) + ggtitle("KEEG of DE genes from sputum with Cluster 5") 

# Disgenet plot() function only displays diseases from ratios > 0.4. 
pclus5Dis <- plot(res_enrich.5, class = "Enrichment",  cutoff= 1) + ggtitle("DisGeNET (sputum) with Cluster 5")

# MSigDb immune signatures
pclus5Msig <- dotplot(ans.tf.5) + ggtitle("MSigDb immunologic signatures")

pAllclus5 <- plot_grid(pclus5GO, pclus5Dis, pclus5KEEG, pclus5Msig, labels = c("A", "B", "C", "D"))

ggsave("Enrichment Cluster 5 TEST.pdf",pAllclus5, width=6, height=7, units="cm", scale=5.5)
```


Cluster 6:

```{r 6_plots}
# GO
pclus6GO <- dotplot(ans.go.6, showCategory=10) + ggtitle("GO of DE genes from sputum associated with Cluster 6") 

# KEGG
x6 <- pairwise_termsim(ans.kegg.6)
pclus6KEEG <- emapplot(x6, showCategory = 10) + ggtitle("KEEG of DE genes from sputum associated with Cluster 6") 

# Disgenet plot() function only displays diseases from ratios > 0.4. 
pclus6Dis <- plot(res_enrich.6, class = "Enrichment",  cutoff= 1) + ggtitle("DisGeNET (sputum) with Cluster 6") 

# MSigDb immune signatures
pclus6Msig <-  dotplot(ans.tf.6) + ggtitle("MSigDb immunologic signatures")

pAllclus6 <- plot_grid(pclus6GO, pclus6Dis, pclus6KEEG, pclus6Msig, labels = c("A", "B", "C", "D"))

ggsave("Enrichment Cluster 6 NEW2.pdf",pAllclus6, width=7, height=7, units="cm", scale=6)
```

